carberretta/bot/cogs/mod.py

```diff
@@ -80,17 +92,153 @@ async def nickname_valid(self, nickname: str) -> bool:
+                    filter_data[item] = replacement
+            except:
+                filter_data[item] = replacement
+
+        async def fix_item_type(item, item_name, checkType, section='mainFilter'):
+            try:
+                if not isinstance(item[item_name], checkType):
+                    return True
+            except:
+                return True
+            return False
+
+        async def fix_item_null(index, item, item_name, replacement, section='mainFilter'):
+            try:
+                if not item[item_name]:
+                    filter_data[section][index][item_name] = replacement
+            except:
+                filter_data[section][index][item_name] = replacement
+
+
+        if path.isfile(self.filter_file):
+            indexes_to_remove = []
+
+            try:
+                async with aiofiles.open(self.filter_file, "r", encoding="utf-8") as f:
+                    filter_data = json.loads(await f.read())
+            except:
+                filter_data = {}
+
+            if not isinstance(filter_data, dict):
+                filter_data = {}
+
+            await fix_category_type('mainFilter', list, [])
+            await fix_category_type('dontFilter', None, None)
+            await fix_category_type('conditionFilter', list, [])
+
+            for index, item in enumerate(filter_data['mainFilter']):
+                if not isinstance(item, dict):
+                    indexes_to_remove.append(index)
+
+                else:
+                    try:
+                        if not item['find'] or not item['word'] or not item['censored']:
+                            indexes_to_remove.append(index)
+                    except:
+                        indexes_to_remove.append(index)
+                    else:
+                        indexes_to_remove.append(index) if await fix_item_type(item, 'find', str) else None
+                        indexes_to_remove.append(index) if await fix_item_type(item, 'word', str) else None
+                        indexes_to_remove.append(index) if await fix_item_type(item, 'censored', str) else None
+
+                        if item['find'] != await self.to_filter_format(item['find']):
+                            filter_data['mainFilter'][index]['find'] = await self.to_filter_format(item['find'])
+
+                        await fix_item_null(index, item, 'added_by', self.bot.user.id)
+                        await fix_item_null(index, item, 'added_on', dt.datetime.utcnow())
+                        await fix_item_null(index, item, 'edited_by', None)
+                        await fix_item_null(index, item, 'edited_on', None)
+
+            for index in indexes_to_remove:
+                filter_data['mainFilter'].pop(index)
+
+            async with aiofiles.open(self.filter_file, "w", encoding="utf-8") as f:
+                await f.write(json.dumps(filter_data, indent=4, cls=chron.DateTimeEncoder))
```

Comment:
The `indent` kwarg isn't necessary unless the user is going to see the file.
```suggestion
                await f.write(json.dumps(filter_data, cls=chron.DateTimeEncoder))
```